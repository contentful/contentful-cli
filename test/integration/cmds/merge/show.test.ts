import nixt from 'nixt'
import { join } from 'path'

const bin = join(__dirname, './../../../../', 'bin')

const app = () => {
  return nixt({ newlines: true }).cwd(bin).base('./contentful.js ').clone()
}

type Result = {
  stderr: string
  stdout: string
}

describe('merge show snapshots', () => {
  it('shows the help properly', done => {
    app()
      .run('merge show --help')
      .code(0)
      .expect(({ stdout }: Result) => {
        const resultText = stdout.trim()
        expect(resultText).toMatchSnapshot('help data is incorrect')
      })
      .end(done)
  })
})

describe('merge show command args validation', () => {
  it('should exit 1 when no args', done => {
    app()
      .run('merge show')
      .code(1)
      .expect(({ stderr }: Result) => {
        const resultText = stderr.trim()
        expect(resultText).toContain('Usage: contentful merge show')
      })
      .end(done)
  })

  it('should exit 1 when no source environment', done => {
    app()
      .run('merge show --te target')
      .code(1)
      .expect(({ stderr }: Result) => {
        const resultText = stderr.trim()
        expect(resultText).toContain('Usage: contentful merge show')
      })
      .end(done)
  })

  it('should exit 1 when no target environment', done => {
    app()
      .run('merge show --se source')
      .code(1)
      .expect(({ stderr }: Result) => {
        const resultText = stderr.trim()
        expect(resultText).toContain('Usage: contentful merge show')
      })
      .end(done)
  })

  it('should exit 1 when no space id passed or in context', done => {
    app()
      .run('merge show --se source --te source')
      .code(1)
      .expect(({ stderr }: Result) => {
        const resultText = stderr.trim()
        expect(resultText).toContain('Error: You need to provide a space id')
      })
      .end(done)
  })

  it('should exit 1 when source and target are the same', done => {
    app()
      .run('merge show --se source --te source --space-id space')
      .code(1)
      .expect(({ stderr }: Result) => {
        const resultText = stderr.trim()
        expect(resultText).toContain(
          'Source and target environments cannot be the same.'
        )
      })
      .end(done)
  })
})
