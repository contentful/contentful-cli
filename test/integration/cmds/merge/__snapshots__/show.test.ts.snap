// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`merge show snapshots shows the help properly: help data is incorrect 1`] = `
"Usage: contentful merge show

Options:
  -h, --help                     Show help                             [boolean]
  --source-environment-id, --se  Source environment id       [string] [required]
  --target-environment-id, --te  Target environment id       [string] [required]
  --yes, -y                      Confirm Merge app installation without prompt"
`;
